scylla-proxy/src/proxy.rs

```diff
@@ -1933,4 +1933,132 @@ mod tests {
                 .expect("Request processing was not concurrent");
         running_proxy.finish().await.unwrap();
     }
+
+    #[tokio::test]
+    #[ntest::timeout(1000)]
+    async fn dry_mode_proxy_drops_incoming_frames() {
+        let node1_proxy_addr = next_local_address_with_port(9876);
+        let proxy = Proxy::new([Node::new_dry_mode(node1_proxy_addr, None)]);
+        let running_proxy = proxy.run().await.unwrap();
+
+        let params = FrameParams {
+            flags: 0,
+            version: 0x04,
+            stream: 0,
+        };
+        let opcode = FrameOpcode::Request(RequestOpcode::Options);
+
+        let body = random_body();
+
+        let mut conn = TcpStream::connect(node1_proxy_addr).await.unwrap();
+
+        write_frame(params, opcode, &body, &mut conn).await.unwrap();
+        // We assert that after sufficiently long time, no error happens inside proxy.
+        tokio::time::sleep(Duration::from_millis(3)).await;
```

Comment:
Note for the future: we should have a way to gracefully close the proxy, i.e. prevent it from accepting more connections and wait until the current connections are closed by the client. This would allow us to avoid such sleeps as above.